### PR TITLE
Changes: synchronize trunk with the 4.05 branch

### PR DESCRIPTION
**Pull request by gasche**

### CODE REVIEW

**Comment by gasche** [comment]
To this Changes-train I added a minor commit to fix the change entry for #1012.

File: Changes [hunk 1]
```diff
@@ -24,26 +24,26 @@ Working version
   bundles.
   (whitequark)
 
-* GPR#659: Remove support for SPARC native code generation
-  (Mark Shinwell)
+- PR#7447, GPR#995: incorrect code generation for nested recursive bindings
```
Comment (Octachron) [inline Changes]: PR#7447 ⇒ MPR#7447?

File: Changes [hunk 2]
```diff
@@ -728,119 +753,109 @@ The complete list of changes is listed below.
...
+- MPR#7504: fix warning 8 with unconstrained records
   (Florian Angeletti, report by John Whitington)
 
-- PR#7456, GPR#1092: fix slow compilation on source files containing a lot
-  of similar debugging information location entries
-  (Mark Shinwell)
+- PR#7511, GPR#1133: Unboxed type with unboxed argument should not be accepted
+  (Damien Doligez, review by Jeremy Yallop and Leo White)
+
+- GPR#795: remove 256-character limitation on Sys.executable_name
```
Comment (Octachron) [inline Changes]: Duplicata at line 335?

**Review by Octachron — COMMENTED** [review]
I found two minor points that could be fixed.

**Comment by gasche** [comment]
@Octachron thanks, I fixed both issues. In fact I converted all PR# mentions in trunk into MPR#, because now that GPR# are four-digit numbers as well I think there are more opportunities for confusion.

File: Changes [hunk 3]
```diff
@@ -24,26 +24,26 @@ Working version
...
   (whitequark)
 
-* GPR#659: Remove support for SPARC native code generation
-  (Mark Shinwell)
+- MPR#7447, GPR#995: incorrect code generation for nested recursive bindings
+  (Leo White and Jeremy Yallop, report by Stephen Dolan)
 
 - MPR#7501, GPR#1089: Consider arrays of length zero as constants
-- PR#7501, GPR#1089: Consider arrays of length zero as constants
+- MPR#7501, GPR#1089: Consider arrays of length zero as constants
```
Comment (damiendoligez) [inline Changes:31]: The header line is duplicated, and the whole entry is duplicated from 4.04.1.

**Review by damiendoligez — APPROVED** [review]
LGTM except for the duplicated entry.

File: Changes (hunk 3 above)
Comment (gasche) [reply Changes:31]: Thanks, fixed. (I think these issues come from a weird git fuzzy merge glitch, at least not from the 4.05 version.). I kept the change in trunk and removed the one in 4.04.1 -- I don't think this change was even available in 4.05.

**Pull request merged**